app/Http/Requests/StoreUserActivityRequest.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+namespace App\Http\Requests;
+
+use Illuminate\Foundation\Http\FormRequest;
+
+class StoreUserActivityRequest extends FormRequest
+{
+    /**
+     * Determine if the user is authorized to make this request.
+     */
+    public function authorize(): bool
+    {
+        // Check if the authenticated user is an admin
+        if ($this->user() && $this->user()->isAdmin()) {
+            return true; // Admins can create records for any users
+        }
+
+        // Check if the 'user_id' in the request matches the authenticated user's ID
+        return $this->user() && $this->user()->id == $this->input('user_id');
```

Comment:
Interesting.. I didn't know about the this->input() method. This would allow for some refactoring of some of the other requests that create another function to call from the controller because this one doesn't allow parameters (for the user_id) as it is called like middleware essentially. Good stuff